docs/explaining_ndcube/metadata.rst

```diff
@@ -0,0 +1,268 @@
+  >>> meta["voltage"]
+  <Quantity [-300., -300., -300., -300., -300.] V>
+
+Unwanted metadata can be removing by employing the ``del`` operator.
+
+.. code-block:: python
+
+  >>> del meta["voltage"]
+  >>> meta.get("voltage", "deleted")
+  'deleted'
+
+Note that the ``del`` operator also removes associated comments and axes.
+
+.. code-block:: python
+
+  >>> meta.key_comments.get("voltage", "deleted")
+  'deleted'
+  >>> meta.axes.get("voltage", "deleted")
+  'deleted'
+
+.. _data_shape:
+
+Data Shape
+----------
+
+The `~ndcube.NDMeta.data_shape` property tracks the shape of the data with which the metadata is associated.
+We have already seen in the :ref:`initializing_ndmeta` section, that it can be assigned during initialization or by subsequently setting the `~ndcube.NDMeta.data_shape` property directly.
+However, if the ``data_shape`` is not provided, it is inferred from the shapes of axis-aware metadata.
+If no axis-aware metadata is present, `~ndcube.NDMeta.data_shape` is empty:
+
+.. code-block:: python
+
+  >>> from ndcube import NDMeta
+  >>> raw_meta = {"salutation": "hello", "name": "world"}
+  >>> meta = NDMeta(raw_meta)
+  >>> meta.data_shape
+  array([], dtype=int64)
+
+If we now add the ``"pixel response"`` metadata that we used, earlier the `~ndcube.NDMeta.data_shape` will be updated.
+
+.. code-block:: python
+
+  >>> meta.add("pixel response", np.array([[100., 85], [100., 100], [100., 90], [90., 100.], [100., 100.]]) * u.percent, axes=(0, 2))
+  >>> meta.data_shape
+  array([5, 0, 2])
+
+Note that since ``"pixel response"`` is associated with the 1st and 3rd axes, those axes now have the same shape as ``"pixel response"``.
+The existence of a 3rd axis, implies the presence of a 2nd.
+However, we have no metadata associated with it, and hence no knowledge of its length.
+It has therefore been assigned a length of ``0``.
+
+Now that the shape has been set for the 1st and 3rd axes, subsequently added grid-aligned metadata associated with those axes must be compatible with those axis lengths.
+For example, if we add a 1-D ``"exposure time"`` and associate it with the 1st axis, it must have a length of of ``5``, otherwise an error will be raised:
+
+.. code-block:: python
+
+  >>> meta.add("exposure time", [1.9, 2.1, 5, 2, 2] * u.s, axes=0)
+
+Moreover, if we now directly set the `~ndcube.NDMeta.data_shape` via ``meta.data_shape = new_shape``, we cannot change the length of axes already associated with grid-aligned metadata, without first removing or altering that metadata.
+However, these restrictions do not apply if we want to change the shape of the 2nd axis, or add new metadata to it, because its length is ``0``, and hence considered undefined.
+
+.. code-block:: python
+
+  >>> meta.add("row temperature", [-10, -11, -12] * u.deg_C, axes=1)
+  >>> meta.data_shape
+  array([5, 3, 2])
+
+.. _assigning_axis_aligned_metadata:
+
+Assigning Axis-aligned Metadata
```

Comment:
"Axis aligned metadata" wasn't a phrase which immediately described what this was to me. My initial thought at the difference between "grid aligned" and "axis aligned" was that grid is aligned to >1 axis and axis aligned was 1.

I am not sure I have a better word, but it's weird.

Reply:
Yeh, I take the point.  I don't have a better phrase either though.